packages/clerk-js/src/ui/utils/createCustomPages.tsx

```diff
@@ -0,0 +1,301 @@
+import { isDevelopmentEnvironment } from '@clerk/shared';
+import type { CustomPage } from '@clerk/types';
+
+import { isValidUrl } from '../../utils';
+import type { NavbarRoute } from '../elements';
+import { CogFilled, TickShield, User } from '../icons';
+import { localizationKeys } from '../localization';
+import { ExternalElementMounter } from './ExternalElementMounter';
+
+export type CustomPageContent = {
+  url: string;
+  mount: (el: HTMLDivElement) => void;
+  unmount: (el?: HTMLDivElement) => void;
+};
+
+type UserProfileReorderItem = {
+  label: 'account' | 'security';
+};
+
+type UserProfileCustomPage = {
+  label: string;
+  url: string;
+  mountIcon: (el: HTMLDivElement) => void;
+  unmountIcon: (el?: HTMLDivElement) => void;
+  mount: (el: HTMLDivElement) => void;
+  unmount: (el?: HTMLDivElement) => void;
+};
```

Comment:
🙃
```suggestion
type UserProfileCustomPage = CustomPage
```

Reply:
This is a different type. `CustomPage` is more generic for all three types of custom items (pages, links, reordering), which is why all of its properties except `label` are optional.